airflow/contrib/example_dags/example_gcp_dataproc_pig_operator.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG for Google Dataproc PigOperator
```

Comment:
I don't think we should add another example dag here - the unit test alone is enough.

Reply:
Example DAGs get "installed" by default in new installs, so we want to be careful about adding example DAGs.

Reply:
All new GCP operators have example DAGs. Existing operators will have examples in this Q. This is necessary because all GCP operators will be tested at the system level. Examples are also used by documentation. This will also be updated this quarter.
I would like to point out that only modules from ``airflow.example_dags`` are automatically loaded. The modules from the ``airflow.contrib.example_dags`` package are not loaded and are not automatically loaded.

Reply:
That is a good point about the 'contrib'.

Do we need an example DAG for every operator - that seems overkill to me.

Reply:
(Useful docs, good. Example DAG for _every_ GCP operator: I'm not so sure)

Reply:
Yeah. @mik-laj  is right. We plan to bring all the GCP operators (even the old ones) up to the level of the new ones we added - including documentation and automated system tests. Those example dags help us to test regressions with real GCP really quickly @ashb. 

We also plan to agree with community and cleanup the contrib vs. main operators and for me this will be great opportunity to move the example dags around to a folder with name that explains that it is mainly for regressions and documentation (they might not be named example_dags any more but something else) @mik-laj - I think we should add it to the proposal of the contrib/main hook synchronisation. @ashb - we are willing to lead the effort, agree to a solution, write proposal and execute most of it for GCP operators at least (possibly for all of them). I think that might be super useful for the Cleanup Summer for 2.0.0 (not Spring any more ;) )

Reply:
This is going a little off topic of this PR now (sorry @nuclearpinguin!) but I personally don't see that integration tests are needed for _every_ dag - for this example the only "interesting" bit is:

```
pig_taks = DataProcPigOperator(
        task_id="pig_task",
        query="define sin HiveUDF('sin');",
        region=REGION,
        cluster_name=CLUSTER_NAME
    )
```

and to test that we just need to check that we call the GCP api correctly - what do we gain from testing this on an actual cluster?

Further: if these dags for for tests and not intended to be visible in a airflow install with `include_examples=True` then they should probably live under `tests/` somewhere.

Reply:
The thing is that we have system tests that start from an empty GCP project, create cluster, run tests on it and then tear it down so that we do not have to pay for it. And we fully automate it in our Cloud Build automated testing. 

In this particular case we will use that example DAG to document all three operators Create/Delete and DataprocOperator itself.. So all three in fact are interesting. We just decided not to do it just yet, but we can just do it now. See for example similar example DAGs we have here: 

https://github.com/apache/airflow/blob/master/airflow/contrib/example_dags/example_gcp_compute.py - we have documentation START/END around every operator and then they are used to have a nice, correct documentation here: https://airflow.readthedocs.io/en/latest/howto/operator/gcp/compute.html  (we know it is correct because we system test it). 

I am going to come back to AIP-4 which was exactly about making this approach (having runnable DAGs as system tests) easy to use by others as well (https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-4+Support+for+System+Tests+for+external+systems). Once AIP-10 and AIP-7 is done, it is very high on my list.

I agree it should not be in "example_dags" possibly. For sure "testable_dags" is much better name - but that's a bigger change that should be part of getting rid of contrib IMHO. 

Maybe as an intermediate step we can simply add the missing documentation and use the example dags in the same way as we did with other operators? So that you can see how useful it is to have it.

Reply:
I think that these examples in the test directory are intended for testing only. These examples are used by the documentation as well.  This, however, requires a deeper thought in a separate ticket.

Reply:
> It is very difficult to perform appropriate tests for all operators without system tests. Unit tests are not enough because we are not able to check that everything is working properly

I agree with @mik-laj. Moreover, some GCP APIs are still in beta phase and system tests allow to catch some inconsistencies and changes.

Reply:
I'm convinced. My main concern was UI "polution" for a fresh install anyway which isn't a problem -- but I think as part of AIP-4 we should think about where these dags live.

Reply:
Yep. I imagine AIP-4 works as plugginable extension to breeze/docker we have in Airflow. Then anyone testing their own services will be able to have their own automated CI running their own system tests using their own CI of choice (for example in our case - using Cloud Build). Without polluting the main Airflow code/tests or examples. I will do it in the way that you might have it living completely outside of the main airflow repo - and you will be able to evolve it separately.

So it all fits very well and I am sure we will find a good resting place for the dags ;).